### DISCUSSION TITLE
Import xterm addons using script tag

### DISCUSSION THREAD

**Discussion by zamoosh** [sourcelair/xterm.js#4621] [Q&A] [answered]
Hello, I'm trying to use the xterm and it's addons like this:

```html
<script src="./xterm/lib/xterm.js"></script>
<script src="./xterm-addon-fit/lib/xterm-addon-fit.js"></script>
```

Initialing xterm is working fine and everything related to it is correct, BUT when I want to include an addon with it, I get errors.

here is the rest of my code:

```html 

<script type="module">

    document.addEventListener("DOMContentLoaded", () => {
        const term = new Terminal({
            screenKeys: true,
            useStyle: true,
            cursorBlink: true,
            fontFamily: "MesloLGS NF, monospace"
        });
        
        const websocket = new WebSocket("ws://" + window.location.hostname + ":" + window.location.port + "/term");
        websocket.binaryType = "arraybuffer";
        
        const fitAddon = new FitAddon();
        term.loadAddon(fitAddon);
        
        term.open(document.getElementById("xterm"));
        
        // ... rest of code ...

        
    });
</script>

``` 

this is my error:
`Uncaught TypeError: FitAddon is not a constructor`

Can someone please help me fix this problem?

here is the code of working terminal:
![image](https://github.com/xtermjs/xterm.js/assets/77716607/18346df3-bf34-4c83-b838-3293aaa7d9ea)

As you can see, it's not fit the screen.

Thanks a million! 🙏🙏

**Comment by jerch** [comment, marked as answer]
Addons are currently bundled as UMD with a namespace object of the same name, e.g. for the fit addon the ctor is located under `FitAddon.FitAddon`. This is needed, as addons may export more than a ctor.

Sidenotes:
- Plz fix the headline, atm it is totally unrelated to your question.
- Your websocket is world readable and prolly writeable - not a good idea. Use `wss://` along with access tokens or other auth mechs. Websockets are tricky when it comes to security, you should read into that or you will create a terminal endpoint accessible for anyone.

**Reply by zamoosh** [reply]
thanks a lot @jerch! this code: `FitAddon.FitAddon` was the answer I want!

- I fixed the headline as you said.
- Yes I know my web socket is not secure, but there is no worry, this is just local test.

Again I say, thanks a lot.
